docs/compilers/CSharp/Compiler Breaking Changes - post VS2017.md

```diff
@@ -1,34 +1,39 @@
-Changes since VS2017 (C# 7)
-===========================
+**This document lists known breaking changes since VS2017 (C# 7)**
 
-- https://github.com/dotnet/roslyn/issues/17089
-In C# 7, the compiler accepted a pattern of the form `dynamic identifier`, e.g. `if (e is dynamic x)`. This was accepted only if the expression `e` was statically of type `dynamic`. The compiler now rejects the use of the type `dynamic` for a pattern variable declaration, as no object's runtime type is `dynamic`.
+<!--
+*Breaking changes are formatted with a numerically delineated list so as to allow shorthand numerical references (e.g., "known break #1").
 
-- https://github.com/dotnet/roslyn/issues/17674 In C# 7, the compiler accepted an assignment statement of the form `_ = M();` where M is a `void` method. The compiler now rejects that.
+Each entry should include a short description of the breaking change, followed by either a link to the issue describing the full details of the change or the full details inline.*
+-->
 
-- https://github.com/dotnet/roslyn/issues/17173 Before C# 7.1, `csc` would accept leading zeroes in the `/langversion` option. Now it should reject it. Example: `csc.exe source.cs /langversion:07`.
+1. In C# 7, the compiler accepted a pattern of the form `dynamic identifier`, e.g. `if (e is dynamic x)`. This was accepted only if the expression `e` was statically of type `dynamic`. The compiler now rejects the use of the type `dynamic` for a pattern variable declaration, as no object's runtime type is `dynamic`.
+https://github.com/dotnet/roslyn/issues/17089 
```

Comment:
It's a nit, but I'd prefer seening MD style links here and throughout this page.:

```suggestion
[dotnet/roslyn#17089](https://github.com/dotnet/roslyn/issues/17089)
```